Test failure: System.Drawing.Tests.SystemFontsTests fail if ran on Korean Windows

Tested environment: Surface Pro 4, running Korean Windows 10 Pro 1809 (Build 17763.316)

Directly from Command prompt, build results:
```
C:\Users\gotos\.nuget\packages\microsoft.dotnet.corefxtesting\1.0.0-beta.19126.3\build\core\Core.targets(190,5): error MSB3073: The command "C:\Users\gotos\source\repos\corefx\artifacts\bin\tests\System.Drawing.Common.Tests\netcoreapp-Windows_NT-Debug-x64\RunTests.cmd C:\Users\gotos\source\repos\corefx\artifacts\bin\testhost/netcoreapp-Windows_NT-Debug-x64/ C:\Users\gotos\source\repos\corefx\.dotnet\ C:\Users\gotos\source\repos\corefx\artifacts\tools\" exited with code 1. [C:\Users\gotos\source\repos\corefx\src\System.Drawing.Common\tests\System.Drawing.Common.Tests.csproj]
C:\Users\gotos\.nuget\packages\microsoft.dotnet.corefxtesting\1.0.0-beta.19126.3\build\core\Core.targets(201,5): error : One or more tests failed while running tests from 'System.Drawing.Common.Tests'. Please check C:\Users\gotos\source\repos\corefx\artifacts\bin\tests\System.Drawing.Common.Tests\netcoreapp-Windows_NT-Debug-x64\testResults.xml for details! [C:\Users\gotos\source\repos\corefx\src\System.Drawing.Common\tests\System.Drawing.Common.Tests.csproj]
```
Relevant parts extracted from testResults.xml:
```xml
<collection total="17" passed="10" failed="7" skipped="0" name="Test collection for System.Drawing.Tests.SystemFontsTests" time="0.037">
      <test name="System.Drawing.Tests.SystemFontsTests.SystemFont_Get_ReturnsExpected_WindowsNames(getFont: Func`1 { Method = System.Drawing.Font &lt;SystemFonts_WindowsNames_TestData&gt;b__2_0(), Target = &lt;&gt;c { } }, systemFontName: \&quot;CaptionFont\&quot;, windowsFontName: \&quot;Segoe UI\&quot;)" type="System.Drawing.Tests.SystemFontsTests" method="SystemFont_Get_ReturnsExpected_WindowsNames" time="0.0068821" result="Fail">
        <traits>
          <trait name="category" value="nonlinuxtests" />
          <trait name="category" value="nonosxtests" />
          <trait name="category" value="nonfreebsdtests" />
          <trait name="category" value="nonnetbsdtests" />
        </traits>
        <failure exception-type="Xunit.Sdk.EqualException">
          <message><![CDATA[Assert.Equal() Failure\r\n          ↓ (pos 0)\r\nExpected: Segoe UI\r\nActual:   맑은 고딕\r\n          ↑ (pos 0)]]></message>
          <stack-trace><![CDATA[   at System.Drawing.Tests.SystemFontsTests.SystemFont_Get_ReturnsExpected_WindowsNames(Func`1 getFont, String systemFontName, String windowsFontName) in C:\Users\gotos\source\repos\corefx\src\System.Drawing.Common\tests\SystemFontsTests.cs:line 74]]></stack-trace>
        </failure>
      </test>
      <test name="System.Drawing.Tests.SystemFontsTests.SystemFont_Get_ReturnsExpected_WindowsNames(getFont: Func`1 { Method = System.Drawing.Font &lt;SystemFonts_WindowsNames_TestData&gt;b__2_1(), Target = &lt;&gt;c { } }, systemFontName: \&quot;IconTitleFont\&quot;, windowsFontName: \&quot;Segoe UI\&quot;)" type="System.Drawing.Tests.SystemFontsTests" method="SystemFont_Get_ReturnsExpected_WindowsNames" time="0.0026561" result="Fail">
        <traits>
          <trait name="category" value="nonlinuxtests" />
          <trait name="category" value="nonosxtests" />
          <trait name="category" value="nonfreebsdtests" />
          <trait name="category" value="nonnetbsdtests" />
        </traits>
        <failure exception-type="Xunit.Sdk.EqualException">
          <message><![CDATA[Assert.Equal() Failure\r\n          ↓ (pos 0)\r\nExpected: Segoe UI\r\nActual:   맑은 고딕\r\n          ↑ (pos 0)]]></message>
          <stack-trace><![CDATA[   at System.Drawing.Tests.SystemFontsTests.SystemFont_Get_ReturnsExpected_WindowsNames(Func`1 getFont, String systemFontName, String windowsFontName) in C:\Users\gotos\source\repos\corefx\src\System.Drawing.Common\tests\SystemFontsTests.cs:line 74]]></stack-trace>
        </failure>
      </test>
      <test name="System.Drawing.Tests.SystemFontsTests.SystemFont_Get_ReturnsExpected_WindowsNames(getFont: Func`1 { Method = System.Drawing.Font &lt;SystemFonts_WindowsNames_TestData&gt;b__2_2(), Target = &lt;&gt;c { } }, systemFontName: \&quot;MenuFont\&quot;, windowsFontName: \&quot;Segoe UI\&quot;)" type="System.Drawing.Tests.SystemFontsTests" method="SystemFont_Get_ReturnsExpected_WindowsNames" time="0.0034311" result="Fail">
        <traits>
          <trait name="category" value="nonlinuxtests" />
          <trait name="category" value="nonosxtests" />
          <trait name="category" value="nonfreebsdtests" />
          <trait name="category" value="nonnetbsdtests" />
        </traits>
        <failure exception-type="Xunit.Sdk.EqualException">
          <message><![CDATA[Assert.Equal() Failure\r\n          ↓ (pos 0)\r\nExpected: Segoe UI\r\nActual:   맑은 고딕\r\n          ↑ (pos 0)]]></message>
          <stack-trace><![CDATA[   at System.Drawing.Tests.SystemFontsTests.SystemFont_Get_ReturnsExpected_WindowsNames(Func`1 getFont, String systemFontName, String windowsFontName) in C:\Users\gotos\source\repos\corefx\src\System.Drawing.Common\tests\SystemFontsTests.cs:line 74]]></stack-trace>
        </failure>
      </test>
      <test name="System.Drawing.Tests.SystemFontsTests.SystemFont_Get_ReturnsExpected_WindowsNames(getFont: Func`1 { Method = System.Drawing.Font &lt;SystemFonts_WindowsNames_TestData&gt;b__2_3(), Target = &lt;&gt;c { } }, systemFontName: \&quot;MessageBoxFont\&quot;, windowsFontName: \&quot;Segoe UI\&quot;)" type="System.Drawing.Tests.SystemFontsTests" method="SystemFont_Get_ReturnsExpected_WindowsNames" time="0.0018576" result="Fail">
        <traits>
          <trait name="category" value="nonlinuxtests" />
          <trait name="category" value="nonosxtests" />
          <trait name="category" value="nonfreebsdtests" />
          <trait name="category" value="nonnetbsdtests" />
        </traits>
        <failure exception-type="Xunit.Sdk.EqualException">
          <message><![CDATA[Assert.Equal() Failure\r\n          ↓ (pos 0)\r\nExpected: Segoe UI\r\nActual:   맑은 고딕\r\n          ↑ (pos 0)]]></message>
          <stack-trace><![CDATA[   at System.Drawing.Tests.SystemFontsTests.SystemFont_Get_ReturnsExpected_WindowsNames(Func`1 getFont, String systemFontName, String windowsFontName) in C:\Users\gotos\source\repos\corefx\src\System.Drawing.Common\tests\SystemFontsTests.cs:line 74]]></stack-trace>
        </failure>
      </test>
      <test name="System.Drawing.Tests.SystemFontsTests.SystemFont_Get_ReturnsExpected_WindowsNames(getFont: Func`1 { Method = System.Drawing.Font &lt;SystemFonts_WindowsNames_TestData&gt;b__2_4(), Target = &lt;&gt;c { } }, systemFontName: \&quot;SmallCaptionFont\&quot;, windowsFontName: \&quot;Segoe UI\&quot;)" type="System.Drawing.Tests.SystemFontsTests" method="SystemFont_Get_ReturnsExpected_WindowsNames" time="0.0019484" result="Fail">
        <traits>
          <trait name="category" value="nonlinuxtests" />
          <trait name="category" value="nonosxtests" />
          <trait name="category" value="nonfreebsdtests" />
          <trait name="category" value="nonnetbsdtests" />
        </traits>
        <failure exception-type="Xunit.Sdk.EqualException">
          <message><![CDATA[Assert.Equal() Failure\r\n          ↓ (pos 0)\r\nExpected: Segoe UI\r\nActual:   맑은 고딕\r\n          ↑ (pos 0)]]></message>
          <stack-trace><![CDATA[   at System.Drawing.Tests.SystemFontsTests.SystemFont_Get_ReturnsExpected_WindowsNames(Func`1 getFont, String systemFontName, String windowsFontName) in C:\Users\gotos\source\repos\corefx\src\System.Drawing.Common\tests\SystemFontsTests.cs:line 74]]></stack-trace>
        </failure>
      </test>
      <test name="System.Drawing.Tests.SystemFontsTests.SystemFont_Get_ReturnsExpected_WindowsNames(getFont: Func`1 { Method = System.Drawing.Font &lt;SystemFonts_WindowsNames_TestData&gt;b__2_5(), Target = &lt;&gt;c { } }, systemFontName: \&quot;StatusFont\&quot;, windowsFontName: \&quot;Segoe UI\&quot;)" type="System.Drawing.Tests.SystemFontsTests" method="SystemFont_Get_ReturnsExpected_WindowsNames" time="0.0019356" result="Fail">
        <traits>
          <trait name="category" value="nonlinuxtests" />
          <trait name="category" value="nonosxtests" />
          <trait name="category" value="nonfreebsdtests" />
          <trait name="category" value="nonnetbsdtests" />
        </traits>
        <failure exception-type="Xunit.Sdk.EqualException">
          <message><![CDATA[Assert.Equal() Failure\r\n          ↓ (pos 0)\r\nExpected: Segoe UI\r\nActual:   맑은 고딕\r\n          ↑ (pos 0)]]></message>
          <stack-trace><![CDATA[   at System.Drawing.Tests.SystemFontsTests.SystemFont_Get_ReturnsExpected_WindowsNames(Func`1 getFont, String systemFontName, String windowsFontName) in C:\Users\gotos\source\repos\corefx\src\System.Drawing.Common\tests\SystemFontsTests.cs:line 74]]></stack-trace>
        </failure>
      </test>
      <test name="System.Drawing.Tests.SystemFontsTests.SystemFont_Get_ReturnsExpected_WindowsNames(getFont: Func`1 { Method = System.Drawing.Font &lt;SystemFonts_WindowsNames_TestData&gt;b__2_6(), Target = &lt;&gt;c { } }, systemFontName: \&quot;DefaultFont\&quot;, windowsFontName: \&quot;Microsoft Sans Serif\&quot;)" type="System.Drawing.Tests.SystemFontsTests" method="SystemFont_Get_ReturnsExpected_WindowsNames" time="0.0043988" result="Fail">
        <traits>
          <trait name="category" value="nonlinuxtests" />
          <trait name="category" value="nonosxtests" />
          <trait name="category" value="nonfreebsdtests" />
          <trait name="category" value="nonnetbsdtests" />
        </traits>
        <failure exception-type="Xunit.Sdk.EqualException">
          <message><![CDATA[Assert.Equal() Failure\r\n          ↓ (pos 0)\r\nExpected: Microsoft Sans Serif\r\nActual:   굴림\r\n          ↑ (pos 0)]]></message>
          <stack-trace><![CDATA[   at System.Drawing.Tests.SystemFontsTests.SystemFont_Get_ReturnsExpected_WindowsNames(Func`1 getFont, String systemFontName, String windowsFontName) in C:\Users\gotos\source\repos\corefx\src\System.Drawing.Common\tests\SystemFontsTests.cs:line 74]]></stack-trace>
        </failure>
      </test>
      <test name="System.Drawing.Tests.SystemFontsTests.SystemFont_Get_ReturnsExpected_WindowsNames(getFont: Func`1 { Method = System.Drawing.Font &lt;SystemFonts_WindowsNames_TestData&gt;b__2_7(), Target = &lt;&gt;c { } }, systemFontName: \&quot;DialogFont\&quot;, windowsFontName: \&quot;Tahoma\&quot;)" type="System.Drawing.Tests.SystemFontsTests" method="SystemFont_Get_ReturnsExpected_WindowsNames" time="0.0101002" result="Pass">
        <traits>
          <trait name="category" value="nonlinuxtests" />
          <trait name="category" value="nonosxtests" />
          <trait name="category" value="nonfreebsdtests" />
          <trait name="category" value="nonnetbsdtests" />
        </traits>
      </test>
      <test name="System.Drawing.Tests.SystemFontsTests.SystemFont_Get_ReturnsExpected(getFont: Func`1 { Method = System.Drawing.Font &lt;SystemFonts_TestData&gt;b__0_0(), Target = &lt;&gt;c { } })" type="System.Drawing.Tests.SystemFontsTests" method="SystemFont_Get_ReturnsExpected" time="0.0012441" result="Pass" />
      <test name="System.Drawing.Tests.SystemFontsTests.SystemFont_Get_ReturnsExpected(getFont: Func`1 { Method = System.Drawing.Font &lt;SystemFonts_TestData&gt;b__0_1(), Target = &lt;&gt;c { } })" type="System.Drawing.Tests.SystemFontsTests" method="SystemFont_Get_ReturnsExpected" time="0.0004829" result="Pass" />
      <test name="System.Drawing.Tests.SystemFontsTests.SystemFont_Get_ReturnsExpected(getFont: Func`1 { Method = System.Drawing.Font &lt;SystemFonts_TestData&gt;b__0_2(), Target = &lt;&gt;c { } })" type="System.Drawing.Tests.SystemFontsTests" method="SystemFont_Get_ReturnsExpected" time="0.0004391" result="Pass" />
      <test name="System.Drawing.Tests.SystemFontsTests.SystemFont_Get_ReturnsExpected(getFont: Func`1 { Method = System.Drawing.Font &lt;SystemFonts_TestData&gt;b__0_3(), Target = &lt;&gt;c { } })" type="System.Drawing.Tests.SystemFontsTests" method="SystemFont_Get_ReturnsExpected" time="0.0004478" result="Pass" />
      <test name="System.Drawing.Tests.SystemFontsTests.SystemFont_Get_ReturnsExpected(getFont: Func`1 { Method = System.Drawing.Font &lt;SystemFonts_TestData&gt;b__0_4(), Target = &lt;&gt;c { } })" type="System.Drawing.Tests.SystemFontsTests" method="SystemFont_Get_ReturnsExpected" time="0.0004856" result="Pass" />
      <test name="System.Drawing.Tests.SystemFontsTests.SystemFont_Get_ReturnsExpected(getFont: Func`1 { Method = System.Drawing.Font &lt;SystemFonts_TestData&gt;b__0_5(), Target = &lt;&gt;c { } })" type="System.Drawing.Tests.SystemFontsTests" method="SystemFont_Get_ReturnsExpected" time="0.0005659" result="Pass" />
      <test name="System.Drawing.Tests.SystemFontsTests.GetFontByName_NoSuchName_ReturnsNull(systemFontName: null)" type="System.Drawing.Tests.SystemFontsTests" method="GetFontByName_NoSuchName_ReturnsNull" time="0.0000956" result="Pass" />
      <test name="System.Drawing.Tests.SystemFontsTests.GetFontByName_NoSuchName_ReturnsNull(systemFontName: \&quot;\&quot;)" type="System.Drawing.Tests.SystemFontsTests" method="GetFontByName_NoSuchName_ReturnsNull" time="0.0000078" result="Pass" />
      <test name="System.Drawing.Tests.SystemFontsTests.GetFontByName_NoSuchName_ReturnsNull(systemFontName: \&quot;captionfont\&quot;)" type="System.Drawing.Tests.SystemFontsTests" method="GetFontByName_NoSuchName_ReturnsNull" time="0.0000058" result="Pass" />
    </collection>
```
(For ease of understanding, `맑은 고딕` is `Malgun Gothic` and `굴림` is `Gulim`)

A few of the tests, specifically `System.Drawing.Tests.SystemFontsTests.SystemFont_Get_ReturnsExpected_WindowsNames` in the SystemFontsTests seem to be failing either due to a fault in test logic or a change from somewhere. The failing assertion is located at [line 74](https://github.com/dotnet/corefx/blob/76e08b605f237bffb74283d83241eaadc4e7ce93/src/System.Drawing.Common/tests/SystemFontsTests.cs#L74), and every test data but the last fails at that line.

I believe it's because Korean Windows do use these fonts as system display fonts, which means that these returned values are actually correct. I do not know if other localised Windows that use fonts other than the ones used by English Windows also fail these tests as well; might worth checking if that's the case.

--- a/src/System.Drawing.Common/tests/SystemFontsTests.cs
+++ b/src/System.Drawing.Common/tests/SystemFontsTests.cs
@@ -3,7 +3,6 @@
 
 using System.Collections.Generic;
 using System.Runtime.InteropServices;
-using Microsoft.VisualStudio.TestPlatform.CommunicationUtilities;
 using Xunit;
 
 namespace System.Drawing.Tests

--- a/src/System.Drawing.Common/tests/SystemFontsTests.cs
+++ b/src/System.Drawing.Common/tests/SystemFontsTests.cs
@@ -85,7 +85,10 @@ namespace System.Drawing.Tests
             Assert.Null(SystemFonts.GetFontByName(systemFontName));
         }
 
-        [DllImport("kernel32.dll", SetLastError = true, CharSet = CharSet.Auto)]
+        [DllImport("kernel32.dll", SetLastError = false, CharSet = CharSet.Auto)]
         public static extern int GetSystemDefaultLCID();
+
+        [DllImport("kernel32.dll", SetLastError = false, CharSet = CharSet.Auto)]
+        public static extern int GetUserDefaultLCID();
     }
 }

--- a/src/System.Drawing.Common/tests/SystemFontsTests.cs
+++ b/src/System.Drawing.Common/tests/SystemFontsTests.cs
@@ -42,59 +42,59 @@ namespace System.Drawing.Tests
 
             switch (userLangId & 0x3ff)
             {
-                case 0x11: //ja-JP (Japanese)
+                case 0x11: // ja-JP (Japanese)
                     fonts = new SystemFontList("Yu Gothic UI");
                     break;
-                case 0x5C: //chr-Cher-US (Cherokee)
+                case 0x5C: // chr-Cher-US (Cherokee)
                     fonts = new SystemFontList("Gadugi");
                     break;
-                case 0x12: //ko-KR (Korean)
+                case 0x12: // ko-KR (Korean)
                     fonts = new SystemFontList("\ub9d1\uc740\x20\uace0\ub515");
                     break;
-                case 0x4: //zh-TW (Traditional Chinese, Taiwan) or zh-CN (Simplified Chinese, PRC)
-                    //Although the primary language ID is the same, the fonts are different
-                    //So we have to determine by the full language ID
-                    //https://docs.microsoft.com/en-us/openspecs/windows_protocols/ms-lcid/70feba9f-294e-491e-b6eb-56532684c37f
-                    //Assuming this doc is correct AND the font only differs by whether it's traditional or not it should work
+                case 0x4: // zh-TW (Traditional Chinese, Taiwan) or zh-CN (Simplified Chinese, PRC)
+                    // Although the primary language ID is the same, the fonts are different
+                    // So we have to determine by the full language ID
+                    // https://docs.microsoft.com/en-us/openspecs/windows_protocols/ms-lcid/70feba9f-294e-491e-b6eb-56532684c37f
+                    // Assuming this doc is correct AND the font only differs by whether it's traditional or not it should work
                     switch (userLangId & 0xFFFF)
                     {
-                        case 0x0004: //zh-Hans
-                        case 0x7804: //zh
-                        case 0x0804: //zh-CN
-                        case 0x1004: //zh-SG
+                        case 0x0004: // zh-Hans
+                        case 0x7804: // zh
+                        case 0x0804: // zh-CN
+                        case 0x1004: // zh-SG
                             fonts = new SystemFontList("Microsoft JhengHei UI");
                             break;
-                        case 0x7C04: //zh-Hant
-                        case 0x0C04: //zh-HK
-                        case 0x1404: //zh-MO
-                        case 0x0404: //zh-TW
+                        case 0x7C04: // zh-Hant
+                        case 0x0C04: // zh-HK
+                        case 0x1404: // zh-MO
+                        case 0x0404: // zh-TW
                             fonts = new SystemFontList("Microsoft YaHei UI");
                             break;
                         default:
                             throw new InvalidOperationException("Cannot determine system fonts");
                     }
                     break;
-                case 0x1E: //th-TH
-                case 0x54: //lo-LA
-                case 0x53: //km-KH 
+                case 0x1E: // th-TH
+                case 0x54: // lo-LA
+                case 0x53: // km-KH 
                     fonts = new SystemFontList("Leelawadee UI");
                     break;
-                case 0x4A: //te-IN
-                case 0x49: //ta-IN
-                case 0x5B: //si-LK
-                case 0x48: //or-IN
-                case 0x4E: //mr-IN
-                case 0x4C: //ml-IN
-                case 0x57: //kok-IN
-                case 0x45: //bn-BD
-                case 0x4D: //as-IN
+                case 0x4A: // te-IN
+                case 0x49: // ta-IN
+                case 0x5B: // si-LK
+                case 0x48: // or-IN
+                case 0x4E: // mr-IN
+                case 0x4C: // ml-IN
+                case 0x57: // kok-IN
+                case 0x45: // bn-BD
+                case 0x4D: // as-IN
                     fonts = new SystemFontList("Nirmala UI");
                     break;
-                case 0x5E: //am-ET
+                case 0x5E: // am-ET
                     fonts = new SystemFontList("Ebrima");
                     break;
-                default: //For now we assume everything else uses Segoe UI
-                    //If there's other failure reported we can add it
+                default: // For now we assume everything else uses Segoe UI
+                    // If there's other failure reported we can add it
                     fonts = new SystemFontList("Segoe UI");
                     break;
             }
@@ -137,8 +137,8 @@ namespace System.Drawing.Tests
         [DllImport("kernel32.dll", SetLastError = false, CharSet = CharSet.Auto)]
         public static extern int GetUserDefaultLCID();
 
-        //Do not test DefaultFont and DialogFont, as we can't reliably determine from LCID
-        //https://github.com/dotnet/corefx/issues/35664#issuecomment-473556522
+        // Do not test DefaultFont and DialogFont, as we can't reliably determine from LCID
+        // https://github.com/dotnet/corefx/issues/35664#issuecomment-473556522
         class SystemFontList
         {
             public SystemFontList(string c_it_m_mb_scFonts)

--- a/src/System.Drawing.Common/tests/SystemFontsTests.cs
+++ b/src/System.Drawing.Common/tests/SystemFontsTests.cs
@@ -3,6 +3,7 @@
 
 using System.Collections.Generic;
 using System.Runtime.InteropServices;
+using Microsoft.VisualStudio.TestPlatform.CommunicationUtilities;
 using Xunit;
 
 namespace System.Drawing.Tests
@@ -162,12 +163,12 @@ namespace System.Drawing.Tests
             {
                 return new []
                 {
-                new object[] {(Func<Font>)(() => SystemFonts.CaptionFont), "CaptionFont", CaptionFont},
-                new object[] {(Func<Font>)(() => SystemFonts.IconTitleFont), "IconTitleFont", IconTitleFont},
-                new object[] {(Func<Font>)(() => SystemFonts.MenuFont), "MenuFont", MenuFont},
-                new object[] {(Func<Font>)(() => SystemFonts.MessageBoxFont), "MessageBoxFont", MessageBoxFont},
-                new object[] {(Func<Font>)(() => SystemFonts.SmallCaptionFont), "SmallCaptionFont", SmallCaptionFont},
-                new object[] {(Func<Font>)(() => SystemFonts.StatusFont), "StatusFont", StatusFont}
+                new object[] {(Func<Font>)(() => SystemFonts.CaptionFont), nameof(CaptionFont), CaptionFont},
+                new object[] {(Func<Font>)(() => SystemFonts.IconTitleFont), nameof(IconTitleFont), IconTitleFont},
+                new object[] {(Func<Font>)(() => SystemFonts.MenuFont), nameof(MenuFont), MenuFont},
+                new object[] {(Func<Font>)(() => SystemFonts.MessageBoxFont), nameof(MessageBoxFont), MessageBoxFont},
+                new object[] {(Func<Font>)(() => SystemFonts.SmallCaptionFont), nameof(SmallCaptionFont), SmallCaptionFont},
+                new object[] {(Func<Font>)(() => SystemFonts.StatusFont), nameof(StatusFont), StatusFont}
                 };
             }
         }

--- a/src/System.Drawing.Common/tests/SystemFontsTests.cs
+++ b/src/System.Drawing.Common/tests/SystemFontsTests.cs
@@ -49,7 +49,7 @@ namespace System.Drawing.Tests
                     fonts = new SystemFontList("Gadugi");
                     break;
                 case 0x12: //ko-KR (Korean)
-                    fonts = new SystemFontList("맑은 고딕");
+                    fonts = new SystemFontList("\ub9d1\uc740\x20\uace0\ub515");
                     break;
                 case 0x4: //zh-TW (Traditional Chinese, Taiwan) or zh-CN (Simplified Chinese, PRC)
                     //Although the primary language ID is the same, the fonts are different

--- a/src/System.Drawing.Common/tests/SystemFontsTests.cs
+++ b/src/System.Drawing.Common/tests/SystemFontsTests.cs
@@ -135,9 +135,6 @@ namespace System.Drawing.Tests
         }
 
         [DllImport("kernel32.dll", SetLastError = false, CharSet = CharSet.Auto)]
-        public static extern int GetSystemDefaultLCID();
-
-        [DllImport("kernel32.dll", SetLastError = false, CharSet = CharSet.Auto)]
         public static extern int GetUserDefaultLCID();
 
         //Do not test DefaultFont and DialogFont, as we can't reliably determine from LCID

--- a/src/System.Drawing.Common/tests/SystemFontsTests.cs
+++ b/src/System.Drawing.Common/tests/SystemFontsTests.cs
@@ -37,21 +37,70 @@ namespace System.Drawing.Tests
 
         public static IEnumerable<object[]> SystemFonts_WindowsNames_TestData()
         {
-            yield return Font(() => SystemFonts.CaptionFont, "CaptionFont", "Segoe UI");
-            yield return Font(() => SystemFonts.IconTitleFont, "IconTitleFont", "Segoe UI");
-            yield return Font(() => SystemFonts.MenuFont, "MenuFont", "Segoe UI");
-            yield return Font(() => SystemFonts.MessageBoxFont, "MessageBoxFont", "Segoe UI");
-            yield return Font(() => SystemFonts.SmallCaptionFont, "SmallCaptionFont", "Segoe UI");
-            yield return Font(() => SystemFonts.StatusFont, "StatusFont", "Segoe UI");
+            int userLangId = GetUserDefaultLCID();
+            SystemFontList fonts;
 
-            bool isArabic = (GetSystemDefaultLCID() & 0x3ff) == 0x0001;
-            yield return Font(() => SystemFonts.DefaultFont, "DefaultFont", isArabic ? "Tahoma" : "Microsoft Sans Serif");
+            switch (userLangId & 0x3ff)
+            {
+                case 0x11: //ja-JP (Japanese)
+                    fonts = new SystemFontList("Yu Gothic UI");
+                    break;
+                case 0x5C: //chr-Cher-US (Cherokee)
+                    fonts = new SystemFontList("Gadugi");
+                    break;
+                case 0x12: //ko-KR (Korean)
+                    fonts = new SystemFontList("맑은 고딕");
+                    break;
+                case 0x4: //zh-TW (Traditional Chinese, Taiwan) or zh-CN (Simplified Chinese, PRC)
+                    //Although the primary language ID is the same, the fonts are different
+                    //So we have to determine by the full language ID
+                    //https://docs.microsoft.com/en-us/openspecs/windows_protocols/ms-lcid/70feba9f-294e-491e-b6eb-56532684c37f
+                    //Assuming this doc is correct AND the font only differs by whether it's traditional or not it should work
+                    switch (userLangId & 0xFFFF)
+                    {
+                        case 0x0004: //zh-Hans
+                        case 0x7804: //zh
+                        case 0x0804: //zh-CN
+                        case 0x1004: //zh-SG
+                            fonts = new SystemFontList("Microsoft JhengHei UI");
+                            break;
+                        case 0x7C04: //zh-Hant
+                        case 0x0C04: //zh-HK
+                        case 0x1404: //zh-MO
+                        case 0x0404: //zh-TW
+                            fonts = new SystemFontList("Microsoft YaHei UI");
+                            break;
+                        default:
+                            throw new InvalidOperationException("Cannot determine system fonts");
+                    }
+                    break;
+                case 0x1E: //th-TH
+                case 0x54: //lo-LA
+                case 0x53: //km-KH 
+                    fonts = new SystemFontList("Leelawadee UI");
+                    break;
+                case 0x4A: //te-IN
+                case 0x49: //ta-IN
+                case 0x5B: //si-LK
+                case 0x48: //or-IN
+                case 0x4E: //mr-IN
+                case 0x4C: //ml-IN
+                case 0x57: //kok-IN
+                case 0x45: //bn-BD
+                case 0x4D: //as-IN
+                    fonts = new SystemFontList("Nirmala UI");
+                    break;
+                case 0x5E: //am-ET
+                    fonts = new SystemFontList("Ebrima");
+                    break;
+                default: //For now we assume everything else uses Segoe UI
+                    //If there's other failure reported we can add it
+                    fonts = new SystemFontList("Segoe UI");
+                    break;
+            }
 
-            bool isJapanese = (GetSystemDefaultLCID() & 0x3ff) == 0x0011;
-            yield return Font(() => SystemFonts.DialogFont, "DialogFont", isJapanese ? "Microsoft Sans Serif" : "Tahoma");
+            return fonts.ToTestData();
         }
-
-        public static object[] Font(Func<Font> getFont, string systemFontName, string windowsFontName) => new object[] { getFont, systemFontName, windowsFontName };
 
         [PlatformSpecific(TestPlatforms.Windows)]
         [ConditionalTheory(Helpers.IsDrawingSupported)]
@@ -90,5 +139,40 @@ namespace System.Drawing.Tests
 
         [DllImport("kernel32.dll", SetLastError = false, CharSet = CharSet.Auto)]
         public static extern int GetUserDefaultLCID();
+
+        //Do not test DefaultFont and DialogFont, as we can't reliably determine from LCID
+        //https://github.com/dotnet/corefx/issues/35664#issuecomment-473556522
+        class SystemFontList
+        {
+            public SystemFontList(string c_it_m_mb_scFonts)
+            {
+                CaptionFont = c_it_m_mb_scFonts;
+                IconTitleFont = c_it_m_mb_scFonts;
+                MenuFont = c_it_m_mb_scFonts;
+                MessageBoxFont = c_it_m_mb_scFonts;
+                SmallCaptionFont = c_it_m_mb_scFonts;
+                StatusFont = c_it_m_mb_scFonts;
+            }
+
+            public string CaptionFont { get; set; }
+            public string IconTitleFont { get; set; }
+            public string MenuFont { get; set; }
+            public string MessageBoxFont { get; set; }
+            public string SmallCaptionFont { get; set; }
+            public string StatusFont { get; set; }
+
+            public IEnumerable<object[]> ToTestData()
+            {
+                return new []
+                {
+                new object[] {(Func<Font>)(() => SystemFonts.CaptionFont), "CaptionFont", CaptionFont},
+                new object[] {(Func<Font>)(() => SystemFonts.IconTitleFont), "IconTitleFont", IconTitleFont},
+                new object[] {(Func<Font>)(() => SystemFonts.MenuFont), "MenuFont", MenuFont},
+                new object[] {(Func<Font>)(() => SystemFonts.MessageBoxFont), "MessageBoxFont", MessageBoxFont},
+                new object[] {(Func<Font>)(() => SystemFonts.SmallCaptionFont), "SmallCaptionFont", SmallCaptionFont},
+                new object[] {(Func<Font>)(() => SystemFonts.StatusFont), "StatusFont", StatusFont}
+                };
+            }
+        }
     }
 }

--- a/src/System.Drawing.Common/tests/SystemFontsTests.cs
+++ b/src/System.Drawing.Common/tests/SystemFontsTests.cs
@@ -71,7 +71,8 @@ namespace System.Drawing.Tests
                             fonts = new SystemFontList("Microsoft YaHei UI");
                             break;
                         default:
-                            throw new InvalidOperationException("Cannot determine system fonts");
+                            throw new InvalidOperationException("The primary language ID is Chinese, however it was not able to" +
+                                                                $" determine the user locale from the LCID with value: {userLangId & 0xFFFF:X4}.");
                     }
                     break;
                 case 0x1E: // th-TH

